### PR TITLE
fix: gsheet save and authorize redirect issue fixed

### LINKED ISSUE #18051: [Bug]-[7372]:save and authorize on auth API navigates to new datasource pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/94514895/199468272-31ae105c-6bb2-4649-bd1f-a849868be857.mov

when I `save and authorize` a new authenticated api datasource successfully, I expect to see the overview of that datasource, but instead I see the new datasources pane.

### Steps To Reproduce

1. create an authenticated api datasource
2. add required fields
3. click `save and authorize`
4. you may see a popup for authorization, click through
Expected: I should see the overview of the datasource

Actual: I see the new datasource pane

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by sneha122**
## Description

When I save and authorize a new authenticated api datasource successfully, I expect to see the overview of that datasource, but instead I see the new datasources pane.

Fixes #18051

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by sumitsum — APPROVED** [review]

**Comment by sumitsum** [comment]
Adding @AmanAgarwal041 to review the client side changes.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/AuthenticatedApiDatasource_spec.js [hunk 1]
```diff
@@ -44,4 +48,28 @@ describe("Authenticated API Datasource", function() {
     cy.contains(queryParams).should("not.exist");
     cy.deleteDatasource("FakeAuthenticatedApi");
   });
+
+  it.only("4. Bug: 18051 - Save and Authorise should return to datasource page in view mode and not new datasource page", () => {
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/AuthenticatedApiDatasource_spec.js:52]: No no I will remove it, I had added it only to run my test. Thanks for noticing this!!

[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/AuthenticatedApiDatasource_spec.js]: Made the changes, can you please check? @Aishwarya-U-R

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good!

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**